x/dex/keeper/liquidity.go

```diff
@@ -97,6 +96,7 @@ func (k Keeper) Swap(
 		), orderFilled, nil
 }
 
+// what's the point of this function if we just write cache every time?
 func (k Keeper) SwapWithCache(
```

Comment:
what's the point of this function if we just write cache every time?

Reply:
Need to confirm but I I think there might be some minor gas savings if we traverse pools multiple times. But it’s largely unnecessary at this point